server-app/src/endpoint/app_controller.py

```diff
@@ -23,11 +24,68 @@ def getTestResponse():
     return "test response"
 
 
-@app.route('/project', methods=['GET'])
+@app.route('/project/all', methods=['GET'])
 def getAllProjects():
     projects = projectDAO.getAllProjects()
     return jsonify({'projects': projects, 'totalSize': len(projects)}), status.HTTP_200_OK
 
+
+@app.route('/project', methods=['GET'])
+def getProjectById():
+    id = request.args.get('id')
+    project = projectDAO.getProjectById(id)
+    return jsonify({'project': project}), status.HTTP_200_OK
+
+
+@app.route('/project', methods=['POST'])
+def createProject():
+    data = request.json
+    id = data.get('id')
```

Comment:
the id should be **generated** here, relying on the client to generate an id which will also be unique is a bit of a long shot. The 2nd best option would be to rely on the database to generate the unique id.

The returned response will contain the object representation as it was inserted into the db (including the new id)